ops/testing.py

```diff
@@ -140,29 +139,47 @@ def _next_relation_id(self):
         self._relation_id_counter += 1
         return rel_id
 
-    def add_relation(self, relation_name, remote_app, remote_app_data=None):
+    def add_relation(self, relation_name, remote_app, *, remote_app_data=None,
+                     initial_unit_data=None, initial_app_data=None):
```

Comment:
Isn't remote_app_data "initial" as well?

The fact we're growing further and further parameters here make me concerned about the fact we expect people to set up the entire scenario in single calls, instead of being able to slowly set up their scenario.

Let's please discuss this live somewhere.

Reply:
Reworked per our discussions:

https://github.com/canonical/operator/pull/212/commits/a8a110a3de064a782e053f99eab2bb592adebfd2